bfgs.go

```diff
@@ -151,10 +151,10 @@ func (b *BFGS) NextDirection(loc *Location, dir []float64) (stepSize float64) {
 	copy(b.grad, loc.Gradient)
 
 	// Compute the new search direction
-	dirmat := mat64.NewDense(b.dim, 1, dir)
-	gradmat := mat64.NewDense(b.dim, 1, loc.Gradient)
+	dirmat := mat64.NewVector(b.dim, dir)
```

Comment:
Could you also rename {dir,grad}mat to probably just d and g? These variables are very local and not matrices anymore. Then LGTM.